org-brain.el

```diff
@@ -62,24 +64,94 @@ This will be used by `org-brain-new-child'."
   :group 'org-brain
   :type '(string))
 
+;;; Utils
+(defun flatten (obj)
+  (do* ((result (list obj))
+        (node result))
+       ((null node) (delete nil result))
+    (cond ((consp (car node))
+           (when (cdar node) (push (cdar node) (cdr node)))
+           (setf (car node) (caar node)))
+          (t (setf node (cdr node))))))
+
+;;; Logging
+(defcustom org-brain-log t
+  "Set to nil to not write to *Messages* buffer."
+  :group 'org-brain
+  :type 'boolean)
+
+(defun org-brain-log (msg)
+  (when org-brain-log
+    (setq inhibit-message t)
+    (message msg)
+    (setq inhibit-message nil)))
+
+;;; Caches
```

Comment:
`defvar` is not constants, so they can be changed. I think a difference is that `defvar` doesn't set the value of the variable if there's already a variable with the same name. This is useful if the user configures the variable before loading the package.